services/venice-server/src/main/java/com/linkedin/venice/listener/request/MultiGetRouterRequestWrapper.java

```diff
@@ -37,16 +36,13 @@ private MultiGetRouterRequestWrapper(
     super(resourceName, keys, isRetryRequest, isStreamingRequest);
   }
 
-  public static MultiGetRouterRequestWrapper parseMultiGetHttpRequest(FullHttpRequest httpRequest) {
-    URI fullUri = URI.create(httpRequest.uri());
-    String path = fullUri.getRawPath();
-    String[] requestParts = path.split("/");
-    if (requestParts.length != 3) {
+  public static MultiGetRouterRequestWrapper parseMultiGetHttpRequest(
+      FullHttpRequest httpRequest,
+      String[] resourceNames) {
```

Comment:
Why not name it `requestParts`, like before? Is it really different resources contained in this array?

Reply:
Sure